tests/lax_numpy_test.py

```diff
@@ -286,6 +286,59 @@ def np_fun(x):
                             atol={dtypes.bfloat16: 1e-1, np.float16: 1e-2})
     self._CompileAndCheck(jnp_fun, args_maker, atol={dtypes.bfloat16: 1e-1})
 
+
+  @jtu.sample_product(
+    [dict(shape=shape, axis=axis)
+      for shape in all_shapes
+      for axis in list(range(-len(shape), len(shape))) + [None] if len(shape) == 1],
+    dtype=all_dtypes,
+    out_dtype=all_dtypes + [None],
+    include_initial=[False, True],
+  )
+  @jax.numpy_dtype_promotion('standard')  # This test explicitly exercises mixed type promotion
+  def testCumulativeSum(self, shape, axis, dtype, out_dtype, include_initial):
+    rng = jtu.rand_some_zero(self.rng())
+    x = rng(shape, dtype)
+    out = jnp.cumulative_sum(x, dtype=out_dtype, include_initial=include_initial)
+
+    target_dtype = out_dtype or x.dtype
+    kind = x.dtype.kind
+    if (out_dtype is None and kind in {'i', 'u'}
+        and x.dtype.itemsize*8 < int(config.default_dtype_bits.value)):
+      target_dtype = dtypes.dtype(dtypes._default_types[kind])
+    assert out.dtype == target_dtype
```

Comment:
We generally use `self._CheckAgainstNumpy` for these kinds of assertions. If `numpy` doesn't have the API yet, we can define a reference function with the correct semantics.